deployment/E_heroku.md

```diff
@@ -1 +1,277 @@
+
+```console
+$ heroku buildpacks:add https://github.com/gjaldon/phoenix-static-buildpack
+Buildpack added. Next release on mysterious-meadow-6277 will use:
+  1. https://github.com/HashNuke/heroku-buildpack-elixir.git
+  2. https://github.com/gjaldon/phoenix-static-buildpack
+Run `git push heroku master` to create a new release using these buildpacks.
+```
+
+## Making our project Heroku-ready
+
+By default, every new Phoenix project ships with a config file `config/prod.secret.exs` that is used to store configuration that should not be commited along with your source code to your repository.
+
+This works great except Heroku [uses environment variables](https://devcenter.heroku.com/articles/config-vars) to pass sensitive informations to our application. It means we need to make some changes to our config before we can deploy.
+
+First, let's copy over the content of `config/prod.secret.exs` to `config/prod.exs`.
+
+Now we need to replace the sensitive values by a tuple `{:system, "MY_CONFIG"}` which will tell Phoenix to load this configuration from an environment variable.
+
+In our case:
+
+```elixir
+config :phoenix_heroku, PhoenixHeroku.Endpoint,
+  secret_key_base: "my_secret_key"
+```
+
+becomes:
+
+```elixir
+config :phoenix_heroku, PhoenixHeroku.Endpoint,
+  secret_key_base: {:system, "SECRET_KEY_BASE"}
+```
+
+and:
+
+```elixir
+config :phoenix_heroku, PhoenixHeroku.Repo,
+  adapter: Ecto.Adapters.Postgres,
+  username: "postgres",
+  password: "postgres",
+  database: "phoenix_heroku_prod",
+  size: 20 # The amount of database connections in the pool
+```
+
+becomes:
+
+```elixir
+config :phoenix_heroku, PhoenixHeroku.Repo,
+  adapter: Ecto.Adapters.Postgres,
+  url: {:system, "DATABASE_URL"},
+  size: 20 # The amount of database connections in the pool
+```
+
+> Note: the use of a tuple instead of a call to `System.get_env/1` is a workaround for releases so that the environment variables are referenced at runtime when using releases. You can read more about it [here](https://github.com/phoenixframework/phoenix/blob/master/lib/phoenix/endpoint.ex#L118).
+
+Also we don't need to import the `config/prod.secret.exs` file in our prod config so you can delete the following line:
+
+```elixir
+import_config "prod.secret.exs"
+```
+
+Your final `config/prod.exs` you should now look like something like this (I've removed the comments for readability):
+
+```
+use Mix.Config
+
+config :phoenix_heroku, PhoenixHeroku.Endpoint,
+  http: [port: {:system, "PORT"}],
+  url: [host: "example.com", port: 80],
+  cache_static_manifest: "priv/static/manifest.json"
+
+config :logger, level: :info
+
+config :phoenix_heroku, PhoenixHeroku.Endpoint,
+  secret_key_base: {:system, "SECRET_KEY_BASE"}
+
+config :phoenix_heroku, PhoenixHeroku.Repo,
```

Comment:
Please switch: `:phoenix_heroku` => `:hello_phoenix` and `PhoenixHeroku` => `HelloPhoenix`